torch/csrc/jit/codegen/cuda/lower_shift.cpp

```diff
@@ -87,6 +82,19 @@ void ShiftPredicateInserter::insert(
   //   }
   // }
 
+  // If the expr involves a thread-block barrier, set the predicate of
+  // the expre with shift_pred. No padding should be necessary.
```

Comment:
note to self: doublecheck if this is really true

Reply:
It should be safe to omit padding. In fact, it's probably safe to omit for all non-shift exprs.